### PR TITLE
feat: project freshness-safe TX target state

### LINKED ISSUE #1968: Project freshness-safe TX target into public runtime state
## Summary

Implement Linear MOR-1106 by projecting canonical `global.tx_state.tx_target` into both public ServerState producer paths without creating a second freshness authority.

## Acceptance criteria

- [ ] Dataclass and StateStore snapshot producers always emit camelCase `txTarget`.
- [ ] Missing target emits `{status:"unknown", reason:"not-observed"}`.
- [ ] Fresh known target maps receiver, slot, and frequency losslessly.
- [ ] A canonically stale target emits `{status:"unknown", reason:"stale"}` and leaks no stale receiver, slot, or frequency.
- [ ] Fresh valid `unsupported` and `contradiction` reasons pass through unchanged.
- [ ] Malformed impossible snapshot values fail closed and never leak partial target data.
- [ ] `fieldStatus.txTarget` reports the canonical store path/freshness coherently and remains diagnostic only.
- [ ] Focused tests cover the reason/freshness matrix and fresh-to-stale transition.

## Scope / non-goals

- `src/rigplane/web/runtime_helpers.py`
- `tests/test_web_runtime_helpers.py`
- Maximum 2 files and 200 net lines.
- Reuse `StateSnapshot` / `FieldSnapshot.freshness` and the merged TxTarget domain.
- No provider acquisition, schema/codegen, frontend selector/component, capability, model, split, selected-RX, or TX lifecycle changes.

## Compatibility

Additive population of the already-staged optional `ServerStatePublic.txTarget`; no CLI, config, rigctld wire, or public command API change.

## Test plan

Focused runtime helper and schema-conformance tests, full non-integration suite, Ruff/format, strict web mypy, import-linter, diff guard, and exact-head CI.

### PR DESCRIPTION
**Pull request by morozsm**
## Summary

- always emit the staged camelCase `txTarget` field from both public-state producer paths
- project fresh canonical StateStore targets losslessly through the merged backend-neutral domain
- replace stale targets with explicit unknown/stale before serialization, preventing receiver/slot/frequency leakage
- keep `fieldStatus.txTarget` on the existing canonical freshness/status path
- fail closed as unknown/contradiction for impossible malformed snapshot values

## Freshness contract

`FieldSnapshot.freshness` is the only freshness authority. Missing data emits unknown/not-observed. Fresh values serialize through the MOR-1047 TxTarget validator. Stale data is discarded before reading target identity and emits only unknown/stale. `fieldStatus.txTarget` is generated from the same snapshot by the existing diagnostic projection and never supplies target identity.

## Scope

- 2 files
- +152 / -1, net +151 LOC
- no provider acquisition, schema/codegen, frontend selector/component, capability, selected-RX, split, model, or TX lifecycle changes

## Validation

- red-first target matrix: 6 expected failures on base
- focused runtime + schema: 58 passed
- focused runtime + schema + web server: 245 passed, 2 skipped
- full non-integration: 8325 passed, 73 skipped, 5 deselected, 11 xfailed, 1 xpassed
- Ruff check and format: passed
- strict web mypy: passed
- import-linter: 5 kept, 0 broken
- state type codegen check: passed
- diff check: passed

Linear: MOR-1106

Closes #1968

### CODE REVIEW

**Comment by morozsm** [comment]
Agent Review: BLOCKED
Head: `7d60661305445f703d258da4b1655a2a7dd385ca`

[P1] `txTarget` and `fieldStatus.txTarget` choose different observations when a `StateSnapshot` contains duplicate entries for `global.tx_state.tx_target`. `_project_snapshot_state_dict` applies tuple order and therefore lets the last entry win (`runtime_helpers.py:929-937`), while `_build_snapshot_field_status` independently chooses the greatest `last_observed_monotonic` (`runtime_helpers.py:454-479`). A newer stale target followed by an older fresh target consequently emits the older known receiver/slot/frequency while diagnostics say the field is stale. That violates the same-snapshot authority requirement and leaks identity across a stale boundary.

Reproduction at this exact head:
```python
fields = (
    FieldSnapshot(path, KnownTxTarget("SUB", "A", 7_074_000), FreshnessState.STALE, 20.0, 1.0, source),
    FieldSnapshot(path, KnownTxTarget("MAIN", "B", 14_074_000), FreshnessState.FRESH, 10.0, 1.0, source),
)
payload = build_public_state_payload_from_snapshot(StateSnapshot(2, 2, 2, 20.0, fields), radio=None, receiver_count=2)
```
Observed: `txTarget == {status:"known", receiver:"MAIN", slot:"B", frequencyHz:14074000}` but `fieldStatus.txTarget.freshness == "stale"`.

Required fix: select one canonical effective field (same timestamp/tie-break rule) for both state projection and diagnostics, or reject duplicate paths before either projection. Add the adversarial duplicate-order matrix in both directions and prove stale always masks identity whenever the chosen diagnostic field is stale. Rerun focused runtime/schema/web checks plus Ruff/format/strict web mypy/import-linter/codegen/diff.

Other reviewed cases passed: missing, fresh known, stale known, stale malformed, unknown freshness, unsupported/contradiction, malformed fresh, input mapping mutation, and JSON serialization. Exact-head quick and grep gates are green.

**Comment by morozsm** [comment]
Remediation pushed at exact head `ad97c6b511151628b667f411b6d1ca3a71a67f77` for the duplicate-path authority split reported in the independent review.

- `tx_target` observations are canonicalized once before both payload projection and `fieldStatus` construction.
- The greatest `last_observed_monotonic` wins independent of snapshot tuple order.
- Equal-time ambiguity is deterministic and fail-closed: stale/unknown outranks fresh; conflicting fresh values project canonical `unknown/contradiction`.
- A stale winner is projected as `unknown/stale` before its value is parsed, so an older fresh identity cannot leak.
- Regressions cover both tuple orders for each scenario, newer stale malformed versus older fresh MAIN/B/14.074 MHz, equal-time freshness/value conflicts, and payload/status consistency.
- Scope remains exactly two owned files, +200/-1 (199 net LOC); no schema, provider, or frontend changes.

Verification on the rebased head:
- `tests/test_web_runtime_helpers.py`: 40 passed
- Full non-integration suite: 8331 passed, 73 skipped, 11 xfailed, 1 xpassed
- Ruff and format check: pass
- strict web mypy: pass
- import-linter: pass
- state type codegen drift check: pass
- diff check: pass

PR intentionally remains Draft and MOR-1106 remains In Progress pending fresh independent review.

**Comment by morozsm** [comment]
Agent Review: PASS
Head: `ad97c6b511151628b667f411b6d1ca3a71a67f77`

Independent re-review confirms the prior duplicate-path authority split is corrected. The snapshot path canonicalizes `global.tx_state.tx_target` once before both payload projection and `fieldStatus` construction. The greatest `last_observed_monotonic` wins independent of tuple order; at equal time stale/unknown fail closed ahead of fresh, and conflicting fresh values become canonical `unknown/contradiction`. Parser spies confirmed stale/unknown winners are not passed to the TxTarget parser, and neither tuple order can leak the older receiver/slot/frequency identity. Payload and diagnostic freshness/availability remained coherent across the adversarial matrix. Missing, fresh known, unsupported, contradiction, malformed fresh, and fresh-to-stale semantics are preserved.

Verified exact scope: 2 files, +200/-1 (199 net LOC).

Independent validation:
- adversarial duplicate-order/freshness/conflict matrix: PASS
- runtime + schema + web: 251 passed, 2 skipped
- full non-integration backend suite: 8543 passed, 73 skipped, 112 deselected, 11 xfailed, 1 xpassed
- Ruff check + format: PASS
- strict web mypy: PASS (25 files)
- import-linter: 5 contracts kept
- state type codegen drift check: PASS
- diff/scope checks: PASS
- exact-head CI: quick and grep-gate PASS

No blocking findings remain.
